### ISSUE TITLE
How to get key point heat maps from COCO Mask R-CNN?

### ISSUE THREAD

**Issue by morgankohler** [facebookresearch/detectron2#754]
I am using the COCO R-CNN model X101-FPN to detect key points in images. The output of this is an instance with the key point location and probability structure with shape (x,y,prob) for each key point. How does COCO Mask R-CNN get key point locations from the heat maps resulting from Mask R-CNN on COCO and how do I get the heat maps back? I tried using the function structures.to_heatmap with the key points as input:
https://detectron2.readthedocs.io/modules/structures.html#detectron2.structures.Keypoints.to_heatmap
I am unsure of what the arguments to this function mean and I am getting strange outputs. I have linked my code and output below.

[heatmap.txt](https://github.com/facebookresearch/detectron2/files/4112229/heatmap.txt)
[heatmap_out.txt](https://github.com/facebookresearch/detectron2/files/4112230/heatmap_out.txt)

**Comment by ppwwyyxx** [comment]
`keypoint_logits` is the predicted heatmap and it is converted to points at this line: https://github.com/facebookresearch/detectron2/blob/2613bf0891e58a734fe9aec0550c8e12608c2b09/detectron2/modeling/roi_heads/roi_heads.py#L714

The output of the model does not contain intermediate heatmap information.
If you need, you can write your own `ROIHeads` and overwrite its `forward_keypoint` method so it does not contain the line above.

**Comment by morgankohler** [comment]
I followed this and was able to extend the structures class to include heat maps. The resulting heatmap is 50 x 50. Is the image being standardized to this size or are the heatmaps subsections of the input image? Thanks for your help.

**Comment by ppwwyyxx** [comment]
It should be 56x56. It's how Keypoing R-CNN works and please refer to the paper for details.

**Comment by morgankohler** [comment]
Is there a list of corresponding joints for the key points list in structures.pred_keypoints? The list is 1 - 17 and I am wondering what joint each index corresponds to. Ex. 1 - left foot, 2 - right foot, etc.

**Comment by ppwwyyxx** [comment]
It's defined by the COCO dataset. It's "nose",
    "left_eye", "right_eye",
    "left_ear", "right_ear",
    "left_shoulder", "right_shoulder",
    "left_elbow", "right_elbow",
    "left_wrist", "right_wrist",
    "left_hip", "right_hip",
    "left_knee", "right_knee",
    "left_ankle", "right_ankle",
